### ISSUE TITLE
Error opening categories with subcats, when a product is marked only for parent category.

### ISSUE THREAD

**Issue by DenisRomanovsky** [bisscomm/refinerycms-products#10]
Steps to reproduce:

1. Create a category.
2. Create a sub. category in it.
3. Create a product. Mark it with parent category. Do not use subcategory.
4. Open parent category page.

Result -- An error happens : 
```
 Showing /home/denis/asdm/app/views/refinery/products/shared/_product.html.erb where line #15 raised:

undefined method `title' for nil:NilClass

  <div class="sub-category">
    <%= product.categories.select{ |c| c.leaf? }.first.title %>
  </div>

  <ul class="properties">
```

Now if to mark the product with sub. category, everything works fine.

Gemfile: 
> gem 'rails', '4.2.1'
> group :development, :test do
> end
>   gem 'mysql2'
> gem 'sass-rails', '~> 5.0'
> gem 'uglifier', '>= 1.3.0'
> gem 'coffee-rails', '~> 4.1.0'
> gem 'therubyracer', platforms: :ruby
> gem 'jquery-rails'
> gem 'turbolinks'
> gem 'jbuilder', '~> 2.0'
> gem 'sdoc', '~> 0.4.0', group: :doc
> group :development, :test do
>   gem 'web-console', '~> 2.0'
> end
> gem 'refinerycms', git: 'https://github.com/refinery/refinerycms', branch: 'master'
> 
> gem 'quiet_assets', :group => :development
> 

> gem 'refinerycms-acts-as-indexed', ['~> 2.0', '>= 2.0.0']
> 
> gem 'refinerycms-wymeditor', ['~> 1.0', '>= 1.0.6']
> 
> gem 'refinerycms-page-images',  path: 'vendor/extensions' #github: 'refinery/refinerycms-page-> images', branch: 'master' #updates taken 04.06.2015
> 
> gem 'refinerycms-products',github: 'bisscomm/refinerycms-products', branch: 'master' #  path: > 'vendor/extensions'  #updates taken 01 July 2015
> 
> #gem 'refinerycms-image_slideshows', git: 'https://github.com/bisscomm/refinerycms-image-> slideshows', branch: 'master'
> 
> gem 'refinerycms-news',  path: 'vendor/extensions' #git: 'https://github.com/refinery/refinerycms-> news', branch: 'master' #updates taken 04.06.2015
> 
> gem 'refinerycms-orders', path: 'vendor/extensions'
> 
> gem 'refinerycms-contacts', path: 'vendor/extensions'
> 
> gem 'refinerycms-search', path: 'vendor/extensions' #github: 'refinery/refinerycms-search', branch: 'master' #updates taken 04.06.2015

Full Trace: 
```
app/views/refinery/products/shared/_product.html.erb:15:in `_app_views_refinery_products_shared__product_html_erb__1899100240769630918_70284757449360'
actionview (4.2.1) lib/action_view/template.rb:145:in `block in render'
activesupport (4.2.1) lib/active_support/notifications.rb:166:in `instrument'
actionview (4.2.1) lib/action_view/template.rb:333:in `instrument'
actionview (4.2.1) lib/action_view/template.rb:143:in `render'
actionview (4.2.1) lib/action_view/renderer/partial_renderer.rb:436:in `block in collection_with_template'
actionview (4.2.1) lib/action_view/renderer/partial_renderer.rb:432:in `map'
actionview (4.2.1) lib/action_view/renderer/partial_renderer.rb:432:in `collection_with_template'
actionview (4.2.1) lib/action_view/renderer/partial_renderer.rb:324:in `render_collection'
actionview (4.2.1) lib/action_view/renderer/partial_renderer.rb:306:in `block in render'
actionview (4.2.1) lib/action_view/renderer/abstract_renderer.rb:39:in `block in instrument'
activesupport (4.2.1) lib/active_support/notifications.rb:164:in `block in instrument'
activesupport (4.2.1) lib/active_support/notifications/instrumenter.rb:20:in `instrument'
activesupport (4.2.1) lib/active_support/notifications.rb:164:in `instrument'
actionview (4.2.1) lib/action_view/renderer/abstract_renderer.rb:39:in `instrument'
actionview (4.2.1) lib/action_view/renderer/partial_renderer.rb:305:in `render'
actionview (4.2.1) lib/action_view/renderer/renderer.rb:47:in `render_partial'
actionview (4.2.1) lib/action_view/renderer/renderer.rb:21:in `render'
actionview (4.2.1) lib/action_view/helpers/rendering_helper.rb:32:in `render'
/home/denis/.rvm/gems/ruby-2.2.1/bundler/gems/refinerycms-products-320aa0bbf83f/app/views/refinery/products/categories/show.html.erb:8:in `block in __home_denis__rvm_gems_ruby_______bundler_gems_refinerycms_products____aa_bbf__f_app_views_refinery_products_categories_show_html_erb___3872089683081875109_70285145457620'
actionview (4.2.1) lib/action_view/helpers/capture_helper.rb:38:in `block in capture'
actionview (4.2.1) lib/action_view/helpers/capture_helper.rb:200:in `with_output_buffer'
actionview (4.2.1) lib/action_view/helpers/capture_helper.rb:38:in `capture'
actionview (4.2.1) lib/action_view/helpers/capture_helper.rb:152:in `content_for'
/home/denis/.rvm/gems/ruby-2.2.1/bundler/gems/refinerycms-products-320aa0bbf83f/app/views/refinery/products/categories/show.html.erb:3:in `__home_denis__rvm_gems_ruby_______bundler_gems_refinerycms_products____aa_bbf__f_app_views_refinery_products_categories_show_html_erb___3872089683081875109_70285145457620'
actionview (4.2.1) lib/action_view/template.rb:145:in `block in render'
activesupport (4.2.1) lib/active_support/notifications.rb:166:in `instrument'
actionview (4.2.1) lib/action_view/template.rb:333:in `instrument'
actionview (4.2.1) lib/action_view/template.rb:143:in `render'
actionview (4.2.1) lib/action_view/renderer/template_renderer.rb:54:in `block (2 levels) in render_template'
actionview (4.2.1) lib/action_view/renderer/abstract_renderer.rb:39:in `block in instrument'
activesupport (4.2.1) lib/active_support/notifications.rb:164:in `block in instrument'
activesupport (4.2.1) lib/active_support/notifications/instrumenter.rb:20:in `instrument'
activesupport (4.2.1) lib/active_support/notifications.rb:164:in `instrument'
actionview (4.2.1) lib/action_view/renderer/abstract_renderer.rb:39:in `instrument'
actionview (4.2.1) lib/action_view/renderer/template_renderer.rb:53:in `block in render_template'
actionview (4.2.1) lib/action_view/renderer/template_renderer.rb:61:in `render_with_layout'
actionview (4.2.1) lib/action_view/renderer/template_renderer.rb:52:in `render_template'
actionview (4.2.1) lib/action_view/renderer/template_renderer.rb:14:in `render'
actionview (4.2.1) lib/action_view/renderer/renderer.rb:42:in `render_template'
actionview (4.2.1) lib/action_view/renderer/renderer.rb:23:in `render'
actionview (4.2.1) lib/action_view/rendering.rb:100:in `_render_template'
actionpack (4.2.1) lib/action_controller/metal/streaming.rb:217:in `_render_template'
actionview (4.2.1) lib/action_view/rendering.rb:83:in `render_to_body'
actionpack (4.2.1) lib/action_controller/metal/rendering.rb:32:in `render_to_body'
actionpack (4.2.1) lib/action_controller/metal/renderers.rb:37:in `render_to_body'
actionpack (4.2.1) lib/abstract_controller/rendering.rb:25:in `render'
actionpack (4.2.1) lib/action_controller/metal/rendering.rb:16:in `render'
actionpack (4.2.1) lib/action_controller/metal/instrumentation.rb:44:in `block (2 levels) in render'
activesupport (4.2.1) lib/active_support/core_ext/benchmark.rb:12:in `block in ms'
/home/denis/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/benchmark.rb:303:in `realtime'
activesupport (4.2.1) lib/active_support/core_ext/benchmark.rb:12:in `ms'
actionpack (4.2.1) lib/action_controller/metal/instrumentation.rb:44:in `block in render'
actionpack (4.2.1) lib/action_controller/metal/instrumentation.rb:87:in `cleanup_view_runtime'
activerecord (4.2.1) lib/active_record/railties/controller_runtime.rb:25:in `cleanup_view_runtime'
actionpack (4.2.1) lib/action_controller/metal/instrumentation.rb:43:in `render'
/home/denis/.rvm/gems/ruby-2.2.1/bundler/gems/refinerycms-29ab03669a0b/pages/lib/refinery/pages/instance_methods.rb:33:in `render_with_presenters'
actionpack (4.2.1) lib/action_controller/metal/implicit_render.rb:10:in `default_render'
actionpack (4.2.1) lib/action_controller/metal/implicit_render.rb:5:in `send_action'
actionpack (4.2.1) lib/abstract_controller/base.rb:198:in `process_action'
actionpack (4.2.1) lib/action_controller/metal/rendering.rb:10:in `process_action'
actionpack (4.2.1) lib/abstract_controller/callbacks.rb:20:in `block in process_action'
activesupport (4.2.1) lib/active_support/callbacks.rb:117:in `call'
activesupport (4.2.1) lib/active_support/callbacks.rb:117:in `call'
activesupport (4.2.1) lib/active_support/callbacks.rb:555:in `block (2 levels) in compile'
activesupport (4.2.1) lib/active_support/callbacks.rb:505:in `call'
activesupport (4.2.1) lib/active_support/callbacks.rb:505:in `call'
activesupport (4.2.1) lib/active_support/callbacks.rb:92:in `_run_callbacks'
activesupport (4.2.1) lib/active_support/callbacks.rb:776:in `_run_process_action_callbacks'
activesupport (4.2.1) lib/active_support/callbacks.rb:81:in `run_callbacks'
actionpack (4.2.1) lib/abstract_controller/callbacks.rb:19:in `process_action'
actionpack (4.2.1) lib/action_controller/metal/rescue.rb:29:in `process_action'
actionpack (4.2.1) lib/action_controller/metal/instrumentation.rb:32:in `block in process_action'
activesupport (4.2.1) lib/active_support/notifications.rb:164:in `block in instrument'
activesupport (4.2.1) lib/active_support/notifications/instrumenter.rb:20:in `instrument'
activesupport (4.2.1) lib/active_support/notifications.rb:164:in `instrument'
actionpack (4.2.1) lib/action_controller/metal/instrumentation.rb:30:in `process_action'
actionpack (4.2.1) lib/action_controller/metal/params_wrapper.rb:250:in `process_action'
activerecord (4.2.1) lib/active_record/railties/controller_runtime.rb:18:in `process_action'
actionpack (4.2.1) lib/abstract_controller/base.rb:137:in `process'
actionview (4.2.1) lib/action_view/rendering.rb:30:in `process'
actionpack (4.2.1) lib/action_controller/metal.rb:196:in `dispatch'
actionpack (4.2.1) lib/action_controller/metal/rack_delegation.rb:13:in `dispatch'
actionpack (4.2.1) lib/action_controller/metal.rb:237:in `block in action'
actionpack (4.2.1) lib/action_dispatch/routing/route_set.rb:74:in `call'
actionpack (4.2.1) lib/action_dispatch/routing/route_set.rb:74:in `dispatch'
actionpack (4.2.1) lib/action_dispatch/routing/route_set.rb:43:in `serve'
actionpack (4.2.1) lib/action_dispatch/journey/router.rb:43:in `block in serve'
actionpack (4.2.1) lib/action_dispatch/journey/router.rb:30:in `each'
actionpack (4.2.1) lib/action_dispatch/journey/router.rb:30:in `serve'
actionpack (4.2.1) lib/action_dispatch/routing/route_set.rb:819:in `call'
railties (4.2.1) lib/rails/engine.rb:518:in `call'
railties (4.2.1) lib/rails/railtie.rb:194:in `public_send'
railties (4.2.1) lib/rails/railtie.rb:194:in `method_missing'
actionpack (4.2.1) lib/action_dispatch/routing/mapper.rb:51:in `serve'
actionpack (4.2.1) lib/action_dispatch/journey/router.rb:43:in `block in serve'
actionpack (4.2.1) lib/action_dispatch/journey/router.rb:30:in `each'
actionpack (4.2.1) lib/action_dispatch/journey/router.rb:30:in `serve'
actionpack (4.2.1) lib/action_dispatch/routing/route_set.rb:819:in `call'
dragonfly (1.0.10) lib/dragonfly/middleware.rb:14:in `call'
dragonfly (1.0.10) lib/dragonfly/middleware.rb:14:in `call'
warden (1.2.3) lib/warden/manager.rb:35:in `block in call'
warden (1.2.3) lib/warden/manager.rb:34:in `catch'
warden (1.2.3) lib/warden/manager.rb:34:in `call'
rack (1.6.4) lib/rack/etag.rb:24:in `call'
rack (1.6.4) lib/rack/conditionalget.rb:25:in `call'
rack (1.6.4) lib/rack/head.rb:13:in `call'
actionpack (4.2.1) lib/action_dispatch/middleware/params_parser.rb:27:in `call'
actionpack (4.2.1) lib/action_dispatch/middleware/flash.rb:260:in `call'
rack (1.6.4) lib/rack/session/abstract/id.rb:225:in `context'
rack (1.6.4) lib/rack/session/abstract/id.rb:220:in `call'
actionpack (4.2.1) lib/action_dispatch/middleware/cookies.rb:560:in `call'
activerecord (4.2.1) lib/active_record/query_cache.rb:36:in `call'
activerecord (4.2.1) lib/active_record/connection_adapters/abstract/connection_pool.rb:649:in `call'
activerecord (4.2.1) lib/active_record/migration.rb:378:in `call'
actionpack (4.2.1) lib/action_dispatch/middleware/callbacks.rb:29:in `block in call'
activesupport (4.2.1) lib/active_support/callbacks.rb:88:in `call'
activesupport (4.2.1) lib/active_support/callbacks.rb:88:in `_run_callbacks'
activesupport (4.2.1) lib/active_support/callbacks.rb:776:in `_run_call_callbacks'
activesupport (4.2.1) lib/active_support/callbacks.rb:81:in `run_callbacks'
actionpack (4.2.1) lib/action_dispatch/middleware/callbacks.rb:27:in `call'
actionpack (4.2.1) lib/action_dispatch/middleware/reloader.rb:73:in `call'
actionpack (4.2.1) lib/action_dispatch/middleware/remote_ip.rb:78:in `call'
actionpack (4.2.1) lib/action_dispatch/middleware/debug_exceptions.rb:17:in `call'
web-console (2.1.3) lib/web_console/middleware.rb:37:in `call'
actionpack (4.2.1) lib/action_dispatch/middleware/show_exceptions.rb:30:in `call'
railties (4.2.1) lib/rails/rack/logger.rb:38:in `call_app'
railties (4.2.1) lib/rails/rack/logger.rb:20:in `block in call'
activesupport (4.2.1) lib/active_support/tagged_logging.rb:68:in `block in tagged'
activesupport (4.2.1) lib/active_support/tagged_logging.rb:26:in `tagged'
activesupport (4.2.1) lib/active_support/tagged_logging.rb:68:in `tagged'
railties (4.2.1) lib/rails/rack/logger.rb:20:in `call'
quiet_assets (1.1.0) lib/quiet_assets.rb:27:in `call_with_quiet_assets'
actionpack (4.2.1) lib/action_dispatch/middleware/request_id.rb:21:in `call'
rack (1.6.4) lib/rack/methodoverride.rb:22:in `call'
rack (1.6.4) lib/rack/runtime.rb:18:in `call'
activesupport (4.2.1) lib/active_support/cache/strategy/local_cache_middleware.rb:28:in `call'
dragonfly (1.0.10) lib/dragonfly/cookie_monster.rb:9:in `call'
rack (1.6.4) lib/rack/lock.rb:17:in `call'
actionpack (4.2.1) lib/action_dispatch/middleware/static.rb:113:in `call'
rack (1.6.4) lib/rack/sendfile.rb:113:in `call'
railties (4.2.1) lib/rails/engine.rb:518:in `call'
railties (4.2.1) lib/rails/application.rb:164:in `call'
rack (1.6.4) lib/rack/lock.rb:17:in `call'
rack (1.6.4) lib/rack/content_length.rb:15:in `call'
rack (1.6.4) lib/rack/handler/webrick.rb:88:in `service'
/home/denis/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/webrick/httpserver.rb:138:in `service'
/home/denis/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/webrick/httpserver.rb:94:in `run'
/home/denis/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/webrick/server.rb:294:in `block in start_thread'
```

**Comment by bricesanchez** [comment]
try this :
```ruby
<%= product.categories.select{ |c| c.leaf? }.first.try(:title) %>
```
or
```ruby
<div class="sub-category">
    <%= if product.categories.select{ |c| c.leaf? } %>
      <%= product.categories.select{ |c| c.leaf? }.first.title %>
    <% else %>
      <%= product.categories.first.title %>
    <% end %>
  </div>
```

It's not really clean but it will help you.

**Comment by DenisRomanovsky** [comment]
This works for me, thank you.

**Comment by bricesanchez** [comment]
Fixed with bae77c87b59f238d3e7e40ba916f4428fb2af85d
